sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,222 @@
+from sympy.physics.mechanics import Body, Lagrangian, KanesMethod, LagrangesMethod
+from sympy.physics.mechanics.method import _Methods
+
+__all__ = ['JointsMethod']
+
+
+class JointsMethod(_Methods):
+    """Method for formulating the equations of motion using a set of interconnected bodies with joints.
+
+    Parameters
+    ==========
+
+    body : Body or ReferenceFrame
+        The inertial(Newtonion) frame.
+    *joints : Joint
+        The joints in the system
+
+    Attributes
+    ==========
+
+    q, u : Matrix
+        Matrices of the generalized coordinates and speeds
+    bodies : iterable
+        Iterable of Body objects in the system.
+    forcelist : iterable
+        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
+        describing the forces on the system.
+    mass_matrix : Matrix
+        The system's mass matrix
+    forcing : Matrix
+        The system's forcing vector
+    mass_matrix_full : Matrix
+        The "mass matrix" for the u's and q's
+    forcing_full : Matrix
+        The "forcing vector" for the u's and q's
```

Comment:
The explanation for this docstring needs to explain more carefully what these matrices and vectors are. Something like:

```
M*u' = F
where u are the generalized speeds

M_full*x' = F_full

where x = [u , q]^T where q are the generalized coordinates
```